### ISSUE TITLE
root 

### ISSUE THREAD

**Issue by jayoh-dev** [cpplint/cpplint#181]
The 'root' option should affect the 'include_subdir' checking.

Here is an source tree example:

```
asset/
source/
- CMakeLists.txt
- a.h
- a.c
test/
```

In the configuration, I would set the root option 'root=source'.
When `a.h` is included in `a.c`, it sould be `#include "a.h"` not `#include "source/a.h".`

But cpplint warn it does not include directory like:

```cpp
#include "a.h" // cpplint warn: "Include the directory when naming .h files  [build/include_subdir]"
```

Regards,
